include/openthread/border_routing.h

```diff
@@ -498,6 +498,20 @@ void otBorderRoutingDhcp6PdSetEnabled(otInstance *aInstance, bool aEnabled);
  */
 otBorderRoutingDhcp6PdState otBorderRoutingDhcp6PdGetState(otInstance *aInstance);
 
+/**
+ * Process a prefix received from the DHCPv6 PD Server. The prefix is received on
+ * the DHCPv6 PD client callback and provided to the Routing Manager via this
+ * API.
+ *
+ * `OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE` must be enabled.
+ * The lifetime of the prefix can be updated by calling the function again with updated
+ * time values.
+ *
+ * @param[in] aInstance   A pointer to an OpenThread instance.
+ * @param[in] aPrefixInfo A pointer to the prefix information structure
+ */
+void otBorderRoutingDhcp6PdProcessPrefix(otInstance *aInstance, otBorderRoutingPrefixTableEntry *aPrefixInfo);
```

Comment:
A suggestion here:

- `otPlatBorderRoutingProcessIcmp6Ra()` (which is the way to inform the DHCP6 prefix) is defined as a platform API (it is in `openthread/platform/border_routing.h`). 
- This new method `otBorderRoutingDhcp6PdProcessPrefix` is a public API (it is in `openthread/border_routing.h`)
- Should we make it consistent? and define the new API as a platform API as well?

Reply:
Yes, I see your point. For me, it seemed that `otBorderRoutingDhcp6PdProcessPrefix` should be a public API but for consistency, I can change it to a platform one. I will update the PR.